In [1]:
# Importing the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# Reading the dataset
train = pd.read_csv('dataset/train.csv')
test = pd.read_csv('dataset/test.csv')

## Getting a look of the dataset

In [3]:
train.head()

Complaint-ID Date-received         Transaction-Type  \
0         Tr-1    11/11/2015                 Mortgage   
1         Tr-2      7/7/2015         Credit reporting   
2         Tr-3      5/7/2015  Bank account or service   
3         Tr-4    11/12/2016          Debt collection   
4         Tr-5     9/29/2016              Credit card   

                           Complaint-reason  \
0  Loan servicing, payments, escrow account   
1    Incorrect information on credit report   
2                 Using a debit or ATM card   
3     Cont'd attempts collect debt not owed   
4                            Payoff process   

                                    Company-response Date-sent-to-company  \
0                                                NaN           11/11/2015   
1   Company chooses not to provide a public response             7/7/2015   
2                                                NaN             5/7/2015   
3  Company believes it acted appropriately as aut...           11/12/2016   
4  Company has responded to the consumer and the ...            9/29/2016   

                  Complaint-Status Consumer-disputes  \
0          Closed with explanation               Yes   
1  Closed with non-monetary relief                No   
2          Closed with explanation                No   
3          Closed with explanation                No   
4          Closed with explanation                No   

                          Consumer-complaint-summary  
0  Seterus, Inc a déposé un faux rapport auprès d...  
1  XX / XX / XXXX La requête en faillite n ° XXXX...  
2  El XXXX / XXXX / 15, estaba preparando el vuel...  
3  The loan was paid in XXXX XXXX. In XXXX, 4 yea...  
4  J'ai obtenu un compte de crédit de soins pour ...

### Dimension of dataset

In [4]:
train.shape

(43266, 9)

So, our dataset has `43266 rows` and `9 columns`. 

<hr>

### Checking the unique values of the columns except the date ones.

In [5]:
# Complaint-ID, looking it for whether all the complaints are unique or there is some category
train['Complaint-ID'].nunique()

43266

In [6]:
# Transaction-Type, number of unique transactions
train['Transaction-Type'].nunique()

18

In [7]:
# Complaint-reason, number of unique reasons
train['Complaint-reason'].nunique()

150

In [8]:
# Company-response, number of unique responses
train['Company-response'].nunique()

10

In [9]:
# Consumer-disputes, number of unique disputes
train['Consumer-disputes'].nunique()

2

In [10]:
# Consumer-complaint-summary, number of unique summaries
train['Consumer-complaint-summary'].nunique()

43022

**There is a need for label encoding in Transaction-type, Complaint-reason, Company-response and Consumer-disputes**

### Summary statistics

In [11]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43266 entries, 0 to 43265
Data columns (total 9 columns):
Complaint-ID                  43266 non-null object
Date-received                 43266 non-null object
Transaction-Type              43266 non-null object
Complaint-reason              43266 non-null object
Company-response              20760 non-null object
Date-sent-to-company          43266 non-null object
Complaint-Status              43266 non-null object
Consumer-disputes             35568 non-null object
Consumer-complaint-summary    43266 non-null object
dtypes: object(9)
memory usage: 3.0+ MB


#### Following are the inferences from above summary:
- Missing values in two columns `Company-response` and `Consumer-disputes`
- Type of columns `Date-received` and `Date-sent-to-company` have the wrong types

In [12]:
train['Date-received'] = pd.to_datetime(train['Date-received'])
train['Date-sent-to-company'] = pd.to_datetime(train['Date-sent-to-company'])

In [14]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43266 entries, 0 to 43265
Data columns (total 9 columns):
Complaint-ID                  43266 non-null object
Date-received                 43266 non-null datetime64[ns]
Transaction-Type              43266 non-null object
Complaint-reason              43266 non-null object
Company-response              20760 non-null object
Date-sent-to-company          43266 non-null datetime64[ns]
Complaint-Status              43266 non-null object
Consumer-disputes             35568 non-null object
Consumer-complaint-summary    43266 non-null object
dtypes: datetime64[ns](2), object(7)
memory usage: 3.0+ MB
